packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -529,19 +543,26 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
 
     override render(): JSX.Element | undefined {
         const { hover } = this.state;
-        const { model, treeNode, colors, parentPath, sourceUri, decoration, labelProvider, commands, menus, contextKeys, caption } = this.props;
+        const { model, treeNode, colors, parentPath, sourceUri, decoration, labelProvider, commands, menus, contextKeys, caption, isLight } = this.props;
         const resourceUri = new URI(sourceUri);
 
+        const [light, dark] = [treeNode.decorations?.icon, treeNode.decorations?.iconDark];
+        const iconTheme = isLight ? light : dark;
+        const cName = typeof iconTheme === 'string' ? iconTheme : '';
+        const isURI = iconTheme && !cName;
+
+        const backgroundImage = isURI ? `url(${iconTheme})` : '';
         const icon = labelProvider.getIcon(resourceUri);
-        const color = decoration && decoration.colorId ? `var(${colors.toCssVariableName(decoration.colorId)})` : '';
-        const letter = decoration && decoration.letter || '';
+        const color = decoration && decoration.colorId && !iconTheme ? `var(${colors.toCssVariableName(decoration.colorId)})` : '';
+        const letter = decoration && decoration.letter && !iconTheme ? decoration.letter : '';
         const tooltip = decoration && decoration.tooltip || '';
         const textDecoration = treeNode.decorations?.strikeThrough === true ? 'line-through' : 'normal';
         const relativePath = parentPath.relative(resourceUri.parent);
         const path = relativePath ? relativePath.fsPath() : labelProvider.getLongName(resourceUri.parent);
         const title = tooltip.length !== 0
             ? `${resourceUri.path.fsPath()} • ${tooltip}`
             : resourceUri.path.fsPath();
+        const ctype = this.isIcon(iconTheme) ? ' decoration-icon' : ' status';
```

Comment:
We should find a better variable name than `ctype` as it is unclear what it is.
I also find it odd why we lead with an empty whitespace.

Reply:
Class name changed.